modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDatabase.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.ingest.geoip;
+
+import com.maxmind.geoip2.model.AsnResponse;
```

Comment:
Since we're making an interface, do we want to take this opportunity to remove the direct dependency on maxmind? Or is it pretty certain that that's the only option we'll ever support?

Reply:
That's a very good question. I think if we were planning on embedding this more widely we would want to provide a facade for the Maxmind libraries so they don't leak out. 

I imagine it wouldn't be too difficult to knock those out for this project. I'm just concerned how much logic is already Maxmind specific in the processor and whether it makes sense to go for it.

Reply:
Things that are Maxmind specific in the GeoIpProcessor right now: 

* The format expected for the database types (e.g. `-City`, `-Country`, `-ASN`)
* Specific logic to remain compliant with the Maxmind database licensing
* The set of default database names for determining if a custom database is specified
* The results we set on ingest documents are based on those available from Maxmind currently. Other database formats would likely need to match capabilities or we'd need to either verify capabilities match the expected fields on the processor or fill the gaps with null values.
* Many error messages mention Maxmind directly

I think without concrete plans for allowing other geo ip database formats, I'd push back on the idea of putting a facade over the Maxmind library unless folks harbor overwhelming support for the idea.